tests/suites/test_suite_ecp.function

```diff
@@ -674,6 +674,52 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void ecp_test_mul( int id, char * n_hex,
+                   char *  Px_hex, char *  Py_hex, char *  Pz_hex,
+                   char * nPx_hex, char * nPy_hex, char * nPz_hex,
+                   int res )
+{
+    mbedtls_ecp_group grp;
+    mbedtls_ecp_point P, nP, R;
+    mbedtls_mpi n;
+    rnd_pseudo_info rnd_info;
+
+    mbedtls_ecp_group_init( &grp ); mbedtls_ecp_point_init( &R );
+    mbedtls_ecp_point_init( &P ); mbedtls_ecp_point_init( &nP );
+    mbedtls_mpi_init( &n );
+    memset( &rnd_info, 0x00, sizeof( rnd_pseudo_info ) );
+
+    TEST_ASSERT( mbedtls_ecp_group_load( &grp, id ) == 0 );
+
+    TEST_ASSERT( mbedtls_ecp_check_pubkey( &grp, &grp.G ) == 0 );
+
+    TEST_ASSERT( mbedtls_mpi_read_string( &n, 16, n_hex ) == 0 );
```

Comment:
What is `16`?

If `16` is what the length of what `n_hex` must be for all test vectors, consider using `data_t n` and `n->x` and `n->len`. (And similar for other parameters)

Reply:
I think 16 is the radix here, ie we want to read this as an hexadecimal number.

Reply:
Perhaps for the benefit on constrained platforms it would be better to use `data_t` and then call `mbedtls_mpi_read_binary()` here. Ie let the test framework to the conversion from hex to binary, and the test body from binary to bignum.